internal/pipe/smtp/smtp.go

```diff
@@ -0,0 +1,96 @@
+package smtp
+
+import (
+	"crypto/tls"
+	"fmt"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/goreleaser/goreleaser/internal/pipe"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/context"
+	gomail "gopkg.in/mail.v2"
+)
+
+const (
+	defaultSubjectTemplate = `{{ .ProjectName }} {{ .Tag }} is out!`
+	defaultBodyTemplate    = `You can view details from: {{ .GitURL }}/releases/tag/{{ .Tag }}`
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string { return "reddit" }
+
+type Config struct {
+	Host     string `env:"SMTP_HOST,notEmpty"`
+	Port     int    `env:"SMTP_PORT,notEmpty"`
+	Username string `env:"SMTP_USERNAME,notEmpty"`
+	Password string `env:"SMTP_PASSWORD,notEmpty"`
+}
+
+func (Pipe) Default(ctx *context.Context) error {
+	if ctx.Config.Announce.SMTP.SubjectTemplate == "" {
+		ctx.Config.Announce.SMTP.SubjectTemplate = defaultSubjectTemplate
+	}
+
+	if ctx.Config.Announce.SMTP.BodyTemplate == "" {
+		ctx.Config.Announce.SMTP.BodyTemplate = defaultBodyTemplate
+	}
+
+	return nil
+}
+
+func (Pipe) Announce(ctx *context.Context) error {
+	if ctx.SkipAnnounce {
+		return pipe.ErrSkipAnnounceEnabled
+	}
+	if !ctx.Config.Announce.SMTP.Enabled {
+		return pipe.ErrSkipDisabledPipe
+	}
+
+	subject, err := tmpl.New(ctx).Apply(ctx.Config.Announce.SMTP.SubjectTemplate)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to SMTP: %w", err)
+	}
+
+	body, err := tmpl.New(ctx).Apply(ctx.Config.Announce.SMTP.BodyTemplate)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to SMTP: %w", err)
+	}
+
+	m := gomail.NewMessage()
+
+	// Set E-Mail sender
+	m.SetHeader("From", ctx.Config.Announce.SMTP.From)
+
+	// Set E-Mail receivers
+	receivers := ctx.Config.Announce.SMTP.To
+	m.SetHeader("To", receivers...)
```

Comment:
should we need to validate given string if its in email format? 🤔

Reply:
I wouldn't... validating email addresses is a pain to get right

Reply:
ok then, I'm resolving the issue by doing nothing.